urdf/_d405.gazebo.xacro

```diff
@@ -1,98 +1,62 @@
 <?xml version="1.0"?>
 <robot name="gazebo_d405" xmlns:xacro="http://www.ros.org/wiki/xacro" xmlns:ignition="http://gazebosim.org/schema">
 
-<!--
-Gazebo plugins can only be included once so this xacro assumes that a parent will include and configure them.
-This xacro requires Gazebo plugin:
-  - ignition::gazebo::systems::Sensors
-  to publish simulated RGBD camera data
+<xacro:include filename="$(find realsense2_gz_description)/urdf/rgb_camera.gazebo.xacro" />
+<xacro:include filename="$(find realsense2_gz_description)/urdf/rgbd_camera.gazebo.xacro" />
 
-For example:
-<gazebo>
-  <plugin filename="libignition-gazebo-sensors-system.so" name="ignition::gazebo::systems::Sensors">
-    <render_engine>ogre2</render_engine>
-  </plugin>
-</gazebo>
+<!-- Xacro args
+    name: should match the name value passed to realsense2_description sensor xacro
+    type:  valid values [rgb or rgbd]. Sensor type you would like to simulate
+    fps: the target frame rate of the camera images
+    gz_topic_name: the topic name the image and camera_info are published to Gazebo topics
+    image_width: See Realsense documentation for what the physical sensor supports
+    image_height: See Realsense documentation for what the physical sensor supports
+    trigger: valid values [true, false]. Only supported with 'type=rgb' and will only publish a single image when triggered on Gazebo topic '${name}/trigger'
+             example command line trigger
+             ign topic -t "/name/trigger" -m Boolean -p "data: true" -n 1
 -->
 <xacro:macro name="gazebo_d405" params="
   name:=camera
+  type:=rgbd
   fps:=15
   gz_topic_name:=camera
   image_width:=1280
-  image_height:=720">
+  image_height:=720
+  triggered:=false">
```

Comment:
my understanding was that there is no trigger option for realsense

if that is the case I think we should not give false hope with this flag being configurable ;)

Reply:
I know but it is a super convenient hack to reduce the compute load if you are willing to deal with the logic of making things work in hardware and simulation.

Next feature I might add is a flag to have gazebo publish [segmented images](https://github.com/MarqRazz/c3pzero/blob/main/c3pzero/c3pzero_description/urdf/c3pzero_kinova_gen3.gazebo.xacro#L62-L77) from the camera for easier vision pipeline emulation without actually having to run a network. Do you think options like these are out of scope for a repo like this?

Reply:
I added some more notes on this to the README.